From a3f70d4675359767e8f138a2ecfc6d540b29d78b Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Tue, 18 Aug 2020 11:07:31 +0100
Subject: [PATCH 2/2] tools: Don't make qemu guess what format the disk image
 is

In autopkgtest-build-qemu, we know vmdb2 produced a raw image; in
autopkgtest-buildvm-ubuntu-cloud, we know that the Ubuntu image is in
qcow2 format, while the cloud-init seed generated by genisoimage is
a raw image.

Signed-off-by: Simon McVittie <smcv@debian.org>
---
 tools/autopkgtest-build-qemu           | 2 +-
 tools/autopkgtest-buildvm-ubuntu-cloud | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/tools/autopkgtest-build-qemu b/tools/autopkgtest-build-qemu
index 9760da1..3d6666a 100755
--- a/tools/autopkgtest-build-qemu
+++ b/tools/autopkgtest-build-qemu
@@ -310,7 +310,7 @@ vmdb2 \
     --image="$image".raw \
     "$vmdb2_config"
 
-qemu-img convert -O qcow2 "$image".raw  "$image".new
+qemu-img convert -f raw -O qcow2 "$image".raw  "$image".new
 
 rm -f "$image".raw
 
diff --git a/tools/autopkgtest-buildvm-ubuntu-cloud b/tools/autopkgtest-buildvm-ubuntu-cloud
index 5b61d86..d396f91 100755
--- a/tools/autopkgtest-buildvm-ubuntu-cloud
+++ b/tools/autopkgtest-buildvm-ubuntu-cloud
@@ -202,7 +202,7 @@ def download_image(cloud_img_url, release, arch):
 
 def resize_image(image, size):
     print('Resizing image, adding %s...' % size)
-    subprocess.check_call(['qemu-img', 'resize', image, '+' + size])
+    subprocess.check_call(['qemu-img', 'resize', '-f', 'qcow2', image, '+' + size])
 
 
 DEFAULT_METADATA = 'instance-id: nocloud\nlocal-hostname: autopkgtest\n'
@@ -339,8 +339,8 @@ def boot_image(image, seed, qemu_command, verbose, timeout):
             '-net', 'user',
             '-net', 'nic,model=virtio',
             '-serial', 'unix:%s,server,nowait' % tty_sock,
-            '-drive', 'file=%s,if=virtio' % image,
-            '-drive', 'file=%s,if=virtio,readonly' % seed]
+            '-drive', 'file=%s,if=virtio,format=qcow2' % image,
+            '-drive', 'file=%s,if=virtio,format=raw,readonly' % seed]
 
     if os.path.exists('/dev/kvm'):
         argv.append('-enable-kvm')
-- 
2.28.0

